cinder_plugin/volume.py

```diff
@@ -114,13 +200,91 @@ def get_attachment(cinder_client, volume_id, server_id):
             return attachment
 
 
+def _get_snapshot_name(ctx, kwargs):
+    return "vol-{}-{}".format(get_openstack_id(ctx), kwargs["snapshot_name"])
+
+
+@with_cinder_client
+def snapshot_create(cinder_client, **kwargs):
+    volume_id = get_openstack_id(ctx)
+
+    backup_name = _get_snapshot_name(ctx, kwargs)
+
+    snapshot_incremental = kwargs["snapshot_incremental"]
```

Comment:
is this a distinct type? Should it be supported in a separate operation?

Reply:
I like to use unified operation "backup" and all specific properties provide by inputs. It will be useful for case when plugin support more than one type of backups.  So we have fully same action for any type of backups.